### ISSUE TITLE
App crashes in background

### ISSUE THREAD

**Issue by lukasdietz** [nextcloud/android#5219]
Hi there,
since a few days my Nextcloud app constantly crashes in background. It happens frequently if the app is not opened.

I am using a OnePlus 3 running HavocOS on Android 10 and Nextcloud Version 3.10.0 RC3.
The error message is always the same:

```
time: 1579197628440
msg: java.lang.SecurityException: Remote stack trace:
	at com.android.server.job.JobServiceContext.assertCallerLocked(JobServiceContext.java:493)
	at com.android.server.job.JobServiceContext.doDequeueWork(JobServiceContext.java:371)
	at com.android.server.job.JobServiceContext$JobCallback.dequeueWork(JobServiceContext.java:160)
	at android.app.job.IJobCallback$Stub.onTransact(IJobCallback.java:169)
	at android.os.Binder.execTransactInternal(Binder.java:1021)

stacktrace: java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$4.done(AsyncTask.java:399)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)
Caused by: java.lang.SecurityException: Caller no longer running, last stopped +4s683ms because: timed out while starting
	at android.os.Parcel.createException(Parcel.java:2071)
	at android.os.Parcel.readException(Parcel.java:2039)
	at android.os.Parcel.readException(Parcel.java:1987)
	at android.app.job.IJobCallback$Stub$Proxy.dequeueWork(IJobCallback.java:292)
	at android.app.job.JobParameters.dequeueWork(JobParameters.java:248)
	at androidx.core.app.JobIntentService$JobServiceEngineImpl.dequeueWork(JobIntentService.java:315)
	at androidx.core.app.JobIntentService.dequeueWork(JobIntentService.java:640)
	at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:390)
	at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:383)
	at android.os.AsyncTask$3.call(AsyncTask.java:378)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 3 more
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.job.JobServiceContext.assertCallerLocked(JobServiceContext.java:493)
	at com.android.server.job.JobServiceContext.doDequeueWork(JobServiceContext.java:371)
	at com.android.server.job.JobServiceContext$JobCallback.dequeueWork(JobServiceContext.java:160)
	at android.app.job.IJobCallback$Stub.onTransact(IJobCallback.java:169)
	at android.os.Binder.execTransactInternal(Binder.java:1021)
```

Hopefully there is somebody to help me with this problem.

**Comment by tobiasKaminsky** [comment]
Hm, I do not see any real issue, other than that the job is running too long.

Maybe this is some strange hardening of HavocOS?

**Comment by The-Compiler** [comment]
I woke up to something similar today:

```************ CAUSE OF ERROR ************

java.lang.RuntimeException: An error occurred while executing doInBackground()
        at android.os.AsyncTask$3.done(AsyncTask.java:354)
        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
        at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)
Caused by: java.lang.SecurityException: Caller no longer running, last stopped +2m52s430ms because: timed out while starting
        at android.os.Parcel.createException(Parcel.java:1950)
        at android.os.Parcel.readException(Parcel.java:1918)
        at android.os.Parcel.readException(Parcel.java:1868)
        at android.app.job.IJobCallback$Stub$Proxy.dequeueWork(IJobCallback.java:195)
        at android.app.job.JobParameters.dequeueWork(JobParameters.java:243)
        at androidx.core.app.JobIntentService$JobServiceEngineImpl.dequeueWork(JobIntentService.java:315)
        at androidx.core.app.JobIntentService.dequeueWork(JobIntentService.java:640)
        at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:390)
        at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:383)
        at android.os.AsyncTask$2.call(AsyncTask.java:333)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        ... 3 more
Caused by: android.os.RemoteException: Remote stack trace:
        at com.android.server.job.JobServiceContext.assertCallerLocked(JobServiceContext.java:481)
        at com.android.server.job.JobServiceContext.doDequeueWork(JobServiceContext.java:359)
        at com.android.server.job.JobServiceContext$JobCallback.dequeueWork(JobServiceContext.java:160)
        at android.app.job.IJobCallback$Stub.onTransact(IJobCallback.java:83)
        at android.os.Binder.execTransact(Binder.java:731)


************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30100190
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: Nokia
Device: NB1
Model: TA-1004
Id: PPR1.180610.011
Product: TA-1004_00WW

************ FIRMWARE ************
SDK: 28
Release: 9
Incremental: 00WW_5_15H
```

**Comment by tobiasKaminsky** [comment]
> Caused by: java.lang.SecurityException: Caller no longer running, last stopped +2m52s430ms because: timed out while starting

This means that between wake up and job execution too much time passes and thus Android is killing it.
What background jobs do you have? Auto upload? Contact upload?

**Comment by lukasdietz** [comment]
In my case I am using Auto Upload and Contact Upload. I am uploading titanium Backups with high file size if this could matter

**Comment by The-Compiler** [comment]
As for my stacktrace above - I'm not using either auto upload nor contact upload, and IIRC there weren't any changes to my Nextcloud content when this happened.

**Comment by tobiasKaminsky** [comment]
Any upload related will be fixed via #5679.

@The-Compiler auto upload scans periodically for new folders to propose them for auto upload.
Maybe this is what (for an unkown reason) causes the crash.
Unfortunately I cannot tel from the stacktrace.

**Comment by tobiasKaminsky** [comment]
Duplicate of #5805
